### ISSUE TITLE
How do I use the cli.py code to get information about the program not working?

### ISSUE THREAD

**Issue by CaliberPlex** [MCMi460/NSO-RPC#45]
This program looks incredibly promising, but I can't get it to work. Obviously saying it doesn't work helps nobody. I'd love to send information that might help fix my issue and improve the program as a whole, but I have absolutely no clue what to do with that [cli.py](https://github.com/MCMi460/NSO-RPC/commit/42be5a344b1bb361dce3634199901c78cadd13d0) code I'm referred to in the FAQ if the program isn't working. Someone mind explaining what I need to do in Layman's terms?

**Comment by MCMi460** [comment]
Yes, I suppose I did not explain it very clearly in the FAQ.
May I ask what the nature of your problem is? Does it simply do nothing when you attempt to run it? And what operating system are you on? Are you able to log in, and then it fails, or can you not even reach that screen? Although running the `app.py` may give you some proper error information, it is unlikely that it will do anything drastically different.
Also, just to be clear: `app.py` is the GUI and `cli.py` is the command-line only tool. I wrote that in the FAQ back before we had a proper GUI, and I just haven't bothered to change it. Sorry for any misconception there could have been.

If you are not using Windows, please tell me and I will change the below instructions to work for that operating system.

<details>
  <summary><b>Instructions on how to run </b><code>app.py</code></summary>
  
To put it simply, go to python.org and download the latest version of Python3.X (10?) for your operating system ([here](https://www.python.org/downloads/)). Using the file downloaded, install Python to your computer. After doing so, go to the main page of this Github repository ([here](https://github.com/MCMi460/NSO-RPC)) and press the big green button that says `Code` and choose the zip file option ([or just click this link, same thing](https://github.com/MCMi460/NSO-RPC/archive/refs/heads/main.zip)). Then, once the zip file of the entire repository is done downloading, extract the file and enter the folder.
You should now see three folders: `scripts`, `resources`, and `client`. Enter the client folder. Now, we are going to do something a bit odd. (I'm assuming you're using Windows. If not, please correct me) In Explorer, click on the bar that has the directory path. It is in the upper third of the window and probably says something like Downloads > NSO-RPC > client. Click on an empty space in that box/bar, typically on the right side (Alternatively, just press ALT+D). It should now allow you to type letters and words in that bar. Type the word `cmd` and press enter. A command prompt will open with your directory selected.
Now, type this:
```python
python -m pip install -r requirements.txt
```
This should install the requirements necessary for NSO-RPC. It may take a minute.
If it fails and gives you some strange error, please comment on this thread about that. If, instead, it says something like "could not find requirements.txt", then you probably didn't enter the client folder, or the command prompt just got misdirectory-ed somehow?
Once that is done, you can run the `app.py` by doing the following command:
```python
python app.py
```

But I must stress again that, if the build you used did not work, there is a great likelihood that the `app.py` will not either.
</details>

I'm sorry for the inconvenience, and thank you for your patience!

**Comment by CaliberPlex** [comment]
> May I ask what the nature of your problem is? Does it simply do nothing when you attempt to run it? And what operating system are you on? Are you able to log in, and then it fails, or can you not even reach that screen?
 
The app prompts me to log in as expected, it logs in just fine and I can see my profile and my friends list, but it always says I'm offline. I tried two games, Splatoon 2 and 3, but nothing changed (if you'd like me to try some other titles, I've got like three dozen games I can try), and I'm on Windows 10. Also on an unrelated note, the options "Select Profile" and "Revert" on the "Profile" tab don't do anything. Dunno if that's related or not. Every other menu element works fine.

> Although running the `app.py` may give you some proper error information, it is unlikely that it will do anything drastically different.

Would you recommend I try running that anyways or would that be a fruitless effort? I can try running it if it would help and isn't too convoluted to figure out.

> I'm sorry for the inconvenience, and thank you for your patience!

No problem at all! Your application is incredibly polished and seems quite operable for the most part, so I'm happy to stick this one through. It's late at night for me so I'll try the instructions tomorrow and get back asap. If there's any further information I can provide that might help please let me know and I'll get back with you tomorrow. Additionally, if you'd like, I can open a separate issue on this project forum as to keep things on subject since the cli.py code isn't relevant to the issue.

Thanks in advance 😄

**Comment by CaliberPlex** [comment]
It's working now! Thanks for the help!
